### PR TITLE
Train class

### PR DESCRIPTION
**Pull request by charlescheung22**
added simple functions

### CODE REVIEW

**Review by zoeyzlee — APPROVED** [review]

**Review by Binhe-Jia — DISMISSED** [review]

**Review by gracelliu — DISMISSED** [review]
lgtm

File: src/main/java/Train.java [hunk 1]
```diff
@@ -6,57 +6,85 @@
...
+    /**
+     *  Either "Online", "Maintenance" or "Offline". Precondition: train in "Maintenance" cannot be online
+     *  Trains that are running can pass Offline trains whether at tracks or at stations. IRL justification: most
+     *  tracks have spare tracks that others can pass
+     */
+    private final HashSet<String> STATUS_TYPES = new HashSet<> (Arrays.asList("Online", "Maintenance", "Offline"));
+    String status;
     int line;  // line number
     boolean direction;  // true for forwards, false for backwards
-    private HashMap<String, ArrayList<Staff>> staff;  // dict mapping from staff type to an arraylist of staff assigned to that role
```
Comment (mattlack15) [inline src/main/java/Train.java:10]: Should use List<> and Map<> interfaces for field types

File: src/main/java/Train.java [hunk 2]
```diff
@@ -6,57 +6,85 @@
...
+import java.util.Arrays;
 import java.util.HashMap;
+import java.util.HashSet;
+import java.util.ArrayList;
 
 class Train {
+
     // constructor type params
-    int capacity;
+    int capacity;  // max capacity
```
Comment (mattlack15) [inline src/main/java/Train.java:17]: Should be private with getter and/or setter

File: src/main/java/Train.java [hunk 3]
```diff
@@ -6,57 +6,85 @@
...
-    //
-
-    private int occupancy = 0;
+    // dict mapping from staff type to an arraylist of staff assigned to that role
+    private HashMap<String, ArrayList<Staff>> staff;
+    private int occupancy;
 
+    // Debatable information can be removed if needed:
+    Node currentLocation;  // assuming that both Station and Track objects can be referenced by this
+    Station nextStation;
```
Comment (mattlack15) [inline src/main/java/Train.java:33]: Consider calculating this instead of having this an actual field

File: src/main/java/Train.java [hunk 4]
```diff
@@ -6,57 +6,85 @@
...
+
     // constructor type params
-    int capacity;
+    int capacity;  // max capacity
+    /**
+     *  Either "Online", "Maintenance" or "Offline". Precondition: train in "Maintenance" cannot be online
+     *  Trains that are running can pass Offline trains whether at tracks or at stations. IRL justification: most
+     *  tracks have spare tracks that others can pass
+     */
+    private final HashSet<String> STATUS_TYPES = new HashSet<> (Arrays.asList("Online", "Maintenance", "Offline"));
```
Comment (mattlack15) [inline src/main/java/Train.java:23]: Should probably be an enum

File: src/main/java/Train.java [hunk 5]
```diff
@@ -6,57 +6,85 @@
...
-     *  @param type:
-     *  @param obj:   TODO: rename obj with better name
+     *  @param position:
+     *  @param employee:
      *  @return boolean whether the staff was successfully added
      */
-    public boolean AddStaff (String type, Staff obj) {
-        if (this.staff.containsKey(type)) {
-            this.staff.get(type).add(obj);
+    public boolean AddStaff (String position, Staff employee) {
```
Comment (mattlack15) [inline src/main/java/Train.java:61]: Naming conventions

File: src/main/java/Train.java [hunk 6]
```diff
@@ -6,57 +6,85 @@
...
         } else {return false;}
     }
 
-    // TODO: add a remover staff function
+    /**
+     *  Query function for staff currently working in a specific position
+     *  @param position: String position to check for the presence of a worker
+     *  @return Arraylist of staff currently in that position, or null if the position was not found
+     */
+    public ArrayList<Staff> GetStaffInJob (String position) {
```
Comment (mattlack15) [inline src/main/java/Train.java:73]: Naming conventions

File: src/main/java/Train.java [hunk 7]
```diff
@@ -6,57 +6,85 @@
...
+        return this.staff.getOrDefault(position, null);
+    }
+
+    /** Setter function to remove staff
+     *
+     *
+     * @return boolean iff the employee was successfully removed
+     */
+    public boolean RemoveStaff (String position, Staff employee) {
```
Comment (mattlack15) [inline src/main/java/Train.java:82]: naming conventions: lowercase first letter

File: src/main/java/Train.java (hunk 3 above)
Comment (mattlack15) [reply src/main/java/Train.java:33]: Referring to nextStation

**Review by zoeyzlee — DISMISSED** [review]

**Review by gracelliu — APPROVED** [review]

**Review by zoeyzlee — APPROVED** [review]

**Review by mattlack15 — APPROVED** [review]

**Review by Binhe-Jia — APPROVED** [review]

**Review by gracelliu — APPROVED** [review]

**Pull request merged**